lib/cli/src/detect.ts

```diff
@@ -112,31 +114,21 @@ export function detectBuilder(packageManager: JsPackageManager) {
     return CoreBuilder.Vite;
   }
 
-  try {
-    let out = '';
-    if (packageManager.type === 'npm') {
-      try {
-        // npm <= v7
-        out = packageManager.executeCommand('npm', ['ls', 'webpack']);
-      } catch (e2) {
-        // npm >= v8
-        out = packageManager.executeCommand('npm', ['why', 'webpack']);
-      }
-    } else {
-      out = packageManager.executeCommand('yarn', ['why', 'webpack']);
+  const webpackVersion = detectWebpack(packageManager);
```

Comment:
The `next` detection seems good but I'm wondering about the Wepback one. With commands like `npm ls` you will get an output like this:

```sh
 ~/ npm ls webpack
next-ts@ /Users/yannbraga/open-source/tmp/next-ts
└─┬ next@12.0.3
  ├─┬ @next/react-dev-overlay@12.0.3
  │ └─┬ webpack@4.46.0
  │   └─┬ terser-webpack-plugin@1.4.5
  │     └── webpack@4.46.0 deduped
  └─┬ @next/react-refresh-utils@12.0.3
    └── webpack@4.46.0 deduped
```

So if any dependency depends on Webpack, it will become a false positive. And in that logic it could happen that the main project does depend on Webpack5 but there's a random lib that depends on Webpack4, making the function return Webpack 4 instead.

Given this info, in Nextjs projects it will actually fall under the webpack check and not reach the nextjs check

Reply:
@yannbf can we flip the order and give nextjs detection priority?

Reply:
Sure that should work. Done!